src/btree/bt_discard.c

```diff
@@ -531,3 +531,19 @@ __wt_free_update_list(WT_SESSION_IMPL *session, WT_UPDATE **updp)
     }
     *updp = NULL;
 }
+
+/*
+ * __wt_free_obsolete_updates --
+ *     Following a globally visible update, free any obsolete updates in the update chain. After a
+ *     globally visible update, no reader finds any updates. It is the responsibility of the caller
+ *     to lock the page before freeing the updates.
+ */
+void
+__wt_free_obsolete_updates(WT_SESSION_IMPL *session, WT_UPDATE *visible_all_upd)
+{
+    WT_UPDATE *next;
+
+    next = visible_all_upd->next;
+    visible_all_upd->next = NULL;
+    __wt_free_update_list(session, &next);
```

Comment:
Should we update the page size here?

Reply:
Thanks. It is a bug that I missed decrementing. I fixed it now.